src/mcode-pilot/utils/serviceResultsProcessing.js

```diff
@@ -0,0 +1,74 @@
+import _ from 'lodash';
+
+/*
+Initialize a row of treatment data. This sets the fields that would be expected by the
+Compass UI to inital default values.
+*/
+function initializeTreatmentData(displayName, treatments) {
+    return {
+        id: _.uniqueId('row_'),
+        displayName,
+        treatments,
+        totalPatients: 0,
+        survivorsPerYear: [],
+        sideEffects: {
+            totalReporting: 0,
+            effects: {}
+        }
+    };
+}
+
+/* Generate a display name for the Compass Data row based off of a set of treatments*/
+function generateTreatmentDisplayName(treatments) {
+    return treatments.map((treatment) => {
+        return treatment.displayName;
+    }).join(" & ");
+}
+
+/* Map an individual service response for a set of treatments into the internal format
+expected by the Compass UI components. */
+function generateOutcomeData(data) {
+    return data.map((item) => {
+        if (item.total === 0 || !item.sufficiency) {
+            return null;
+        }
+        let row = initializeTreatmentData(generateTreatmentDisplayName(item.treatments), item.treatments);
+        row.totalPatients = item.total;
+
+        if (item.sideEffects !== undefined) {
```

Comment:
Recommend using `item.sideEffects != undefined` or `!= null` here (and similar comments in other places where `!== undefined` is currently used. This variable may never be `null` in the code as written today but it's safe to be a little defensive here.

Reply:
Can we use `_.isEmpty` or something along those lines?